omegaconf/_utils.py

```diff
@@ -189,6 +190,19 @@ def _resolve_optional(type_: Any) -> Tuple[bool, Any]:
     return False, type_
 
 
+def _is_optional(obj: Any, key: Optional[Union[int, str]] = None) -> bool:
+    """Check `obj` metadata to see if the given node is optional."""
+    from .base import Container, Node
+
+    if key is not None:
+        assert isinstance(obj, Container)
+        obj = obj._get_node(key)
+    if isinstance(obj, Node):
+        return obj._is_optional()
+    else:
+        return True
```

Comment:
now that we have a separate implementation for the internal one, is there a need to support objects that are not nodes?

Reply:
There is not.
But the `else` block still catches the case where `key` cannot be found.
What do you think of raising `ConfigKeyError` if `key` is not found, instead of returning True? (see my https://github.com/omry/omegaconf/pull/700#discussion_r623369771).

Reply:
missed your other question. answered there.
resolving.